factcast-store/src/main/java/org/factcast/store/internal/telemetry/PgStoreTelemetry.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2017-2024 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.internal.telemetry;
+
+import com.google.common.eventbus.AsyncEventBus;
+import com.google.common.eventbus.EventBus;
+import java.util.concurrent.Executors;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.factcast.core.subscription.SubscriptionRequestTO;
+
+@Slf4j
+public final class PgStoreTelemetry {
+
+  private final EventBus eventBus;
+
+  public PgStoreTelemetry() {
+    this(
+        new AsyncEventBus(
+            PgStoreTelemetry.class.getSimpleName(),
+            // needs to be a singleThread executor to ensure that the order of events is
+            // respected
+            Executors.newSingleThreadExecutor()));
```

Comment:
I wonder if it would be valuable to instrument the ES via micrometer. We could see some metrics in our dashboards about how many signals are dispatched/waiting times. Same could be done for the old eventbus. Wdyt?

Reply:
Yes that would definitely make sense. It would be a lot easier to troubleshoot possible bottlenecks.